rnote-ui/src/workspacebrowser/mod.rs

```diff
@@ -163,64 +163,224 @@ impl WorkspaceBrowser {
     }
 
     pub fn init(&self, appwindow: &RnoteAppWindow) {
-        let remove_workspace_button = self.imp().remove_workspace_button.get();
-        let edit_workspace_button = self.imp().edit_workspace_button.get();
+        setup_remove_workspace_button(self, appwindow);
+        setup_add_workspace_button(self, appwindow);
+        setup_edit_workspace_button(self, appwindow);
 
-        self.imp().add_workspace_button.get().connect_clicked(
-            clone!(@weak self as workspacebrowser, @weak appwindow => move |_add_workspace_button| {
-                let dir = workspacebrowser.selected_workspace_dir().unwrap_or(PathBuf::from("./"));
-                workspacebrowser.add_workspace(dir);
+        {
+            let remove_workspace_button = self.imp().remove_workspace_button.get();
+            let edit_workspace_button = self.imp().edit_workspace_button.get();
+            setup_workspacelist(
+                self,
+                appwindow,
+                remove_workspace_button,
+                edit_workspace_button,
+            );
+        }
```

Comment:
Tbh, I'm a bit unsure if we should remove those `let` lines and move them inside to `setup_workspacelist` or not.

On the one hand you can clearly see here, that those two widgets are needed here but on the other hand: Is it necessary to see in the init function that the workspacelist needs those two widgets? What do you think?

Reply:
I would just move them into `setup_workspacelist()`. It is very cheap to call `imp()` and `.get()`, so it can be done in multiple places wherever one needs a reference to the widgets

Reply:
Ayeaye captain!

Reply:
done